app/components/App.js

```diff
@@ -44,18 +48,23 @@ class App extends React.Component {
 		this.handleClickOnBoard = this.handleClickOnBoard.bind(this);
 		this.handleStartClick = this.handleStartClick.bind(this);
 		this.handleEndClick = this.handleEndClick.bind(this);
+		this.handleGoClick = this.handleGoClick.bind(this);
 		this.updateStartLocation = this.updateStartLocation.bind(this);
 		this.updateEndLocation = this.updateEndLocation.bind(this);
+		this.updatePath = this.updatePath.bind(this);
 	}
 
 	handleClickOnBoard(x, y) {
 		console.log('trying to update state');
+		const board = this.state.board;
 		if (this.state.startFlag) {
 			this.updateStartLocation(x, y);
 		} else if (this.state.endFlag) {
 			this.updateEndLocation(x, y);
 		} else {
-			const tempBoard = this.state.board;
+			const tempBoard = board.map((row) =>{
+				return row.map((cell)=>Object.assign({}, cell));
```

Comment:
because I want to make changes to state only from `this.setState()`, and since `board` is a nested array of objects, if i make a change to the cell's content it will change the state without invoking the `setState` function.